src/reject.rs

```diff
@@ -362,7 +362,6 @@ impl fmt::Debug for Reason {
 }
 
 #[doc(hidden)]
-#[deprecated(note = "Use warp::reply::json and warp::reply::with_status instead.")]
```

Comment:
you refer that new compiler gives a deprecation warning but I was not able to reproduce with `1.52` nor `nightly`

Reply:
Thanks for taking a look! This is what I see with deprecated part:
```
   Compiling warp v0.1.23 (/home/user/www/warp)
error: this `#[deprecated]` annotation has no effect
   --> /home/ubuntu/www/warp/src/reject.rs:365:1
    |
365 | #[deprecated(note = "Use warp::reply::json and warp::reply::with_status instead.")]
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: remove the unnecessary deprecation attribute
    |
    = note: `#[deny(useless_deprecated)]` on by default

error: aborting due to previous error

error: could not compile `warp`
```
```
cargo --version
cargo 1.52.0 (69767412a 2021-04-21)
```

Reply:
This starts to fail with cargo 1.49.0 (d00d64df9 2020-12-05), 1.48 still works while compiling [mles-webproxy](https://crates.io/crates/mles-webproxy)-crate.